src/platform/linux_usbfs/usbfs.rs

```diff
@@ -71,15 +70,43 @@ struct UsbFsIoctl {
     data: *mut c_void,
 }
 
+/// Opcodes used in ioctl with the usb device fs.
+///
+/// Taken from https://github.com/torvalds/linux/blob/e9680017b2dc8686a908ea1b51941a91b6da9f1d/include/uapi/linux/usbdevice_fs.h#L187
+// TODO: Move the rest of the opcodes into here?
+#[allow(non_camel_case_types)]
+mod opcodes {
+    use super::*;
+
+    // We repeat the USBDEVFS_ prefix to help keep the same names as what linux uses.
+    // This makes the code more searchable.
+
+    pub type USBDEVFS_IOCTL = ioctl::ReadWriteOpcode<b'U', 18, UsbFsIoctl>;
+    pub type USBDEVFS_DISCONNECT = ioctl::NoneOpcode<b'U', 22, ()>;
+    pub type USBDEVFS_CONNECT = ioctl::NoneOpcode<b'U', 23, ()>;
+    pub type USBDEVFS_DISCONNECT_CLAIM = ioctl::ReadOpcode<b'U', 27, DetachAndClaim>;
```

Comment:
>  I am unfamiliar with how the kernel attaches drivers to devices and never used the usbfs before. I have also never used ioctl before.

This is a really confusing place to start because there are two levels of ioctls here. `USBDEVFS_IOCTL` and `USBDEVFS_DISCONNECT_CLAIM` are "real" ioctl opcodes passed directly to the `ioctl` syscall. Then, for some reason `USBDEVFS_IOCTL` takes parameters that are structured as if they were ioctl opcodes  for `USBDEVFS_DISCONNECT` and `USBDEVFS_CONNECT`.

So even though the kernel header puts those two in the same list as the others, it probably makes more sense to separate them because they are used differently.

Reply:
I created a `nested` module, let me know if that works.